sys/usb/usbus/msc/msc.c

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (C) 2019-2021 Mesotic SAS
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup usbus_msc
+ * @{
+ *
+ * @author  Dylan Laduranty <dylan.laduranty@mesotic.com>
+ * @}
+ */
+
+#include "usb/descriptor.h"
+#include "usb/usbus.h"
+#include "usb/usbus/control.h"
+
+#include "usb/msc.h"
+#include "usb/usbus/msc.h"
+#include "usb/usbus/msc/scsi.h"
+#include "board.h"
+
+#include <string.h>
+
+#define ENABLE_DEBUG    0
+#include "debug.h"
+
+#include "mtd.h"
+extern mtd_dev_t *mtd0;
+
+/* Internal buffer to handle size difference between MTD layer and USB
+   endpoint size as some MTD implementation (like mtd_sdcard) doesn't allow
+   endpoint size transfer */
```

Comment:
Do you mean it does not support reading at an offset?
`mtd_sdcard` might actually be the only driver that does not support that.

Reply:
Yes mtd_sdcard doesn't support the offset argument so I had to workaround this.

Reply:
[`mtd_mci`](https://github.com/RIOT-OS/RIOT/blob/master/drivers/mtd_mci/mtd_mci.c), which is also used to drive SD-cards, has an internal page buffer to support offsets.

I wonder if we should rather add that to `mtd_sdcard` too (maybe as a module, so it can be left out if e.g. fatfs is used, that will only ever write pages and has it's own page buffer for that.

Reply:
If there is a way to use the offset for sdcard, I would be happy to adapt the code.
Nevertheless, I think we should keep an internal buffer to store a page in memory for write operations to avoid multiple write (64 bytes) for a single page.